File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        nums=[(num,i) for i,num in enumerate(nums)]
        nums.sort()
        l=0
        r=len(nums)-1
        while l!=r:
            if nums[l][0]+nums[r][0]<target:
                l+=1
            elif nums[l][0]+nums[r][0]>target:
                r-=1
            else:
                return [nums[l][1],nums[r][1]]

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> m=new HashMap<>();
        int i=0;
        for(int n:nums){
            int r=target-n;
            if(m.containsKey(r)){
                return new int[]{m.get(r),i};
            }
            m.put(n,i);
            i+=1;
        }
        return new int[]{};
    }
}